rename.ml

```diff
@@ -21,6 +21,15 @@ let fresh_label instrs label =
     then cand_lab else find (i+1) in
   find 1
 
+let fresh_version_label (prog:program) label =
+  let cand i = label ^ "_" ^ (string_of_int i) in
+  let existing = fst (List.split prog) in
+  let rec find i =
+    let cand_lab = cand i in
+    if not (List.mem cand_lab existing)
+    then cand_lab else find (i+1) in
+  find 1
```

Comment:
We could have a `fresh_in_list : string list -> string -> string` function to avoid repeating the concatenation logic each time.